### PR TITLE
Update p2p to handle new invs/messages

### PR DESCRIPTION
**Pull request by UdjinM6**
~Also disables "Unneeded inv type"  log spam.~

### CODE REVIEW

File: p2pool/util/pack.py [hunk 1]
```diff
@@ -192,6 +192,13 @@ def read(self, file):
     def write(self, file, item):
         return file, struct.pack(self.desc, item)
 
+class BoolType(StructType):
+    # self.announce = struct.unpack("<?", f.read(1))[0]
```
Comment (codablock) [inline p2pool/util/pack.py]: What are these comments for? Leftovers from testing?

File: p2pool/dash/p2p.py [hunk 2]
```diff
@@ -106,9 +106,9 @@ def handle_inv(self, invs):
                 self.factory.new_block.happened(inv['hash'])
             elif inv['type'] == 'tx':
                 self.send_getdata(requests=[inv])
-            else:
-                if p2pool.DEBUG:
```
Comment (codablock) [inline p2pool/dash/p2p.py]: This logging was only active in debug mode, so it shouldn't bother most users. As you have updated the known inv types, the log should be silenced now nevertheless. If log entries would reappear, people would know that stuff is missin gin the known inv list.

So the question is...should we really remove the logging here? We should either keep the inv type list up-to-date, or ignore unknown inv types. Doing both won't work IMHO (as we'll not get notified about missing inv types).

File: p2pool/util/pack.py (hunk 1 above)
Comment (UdjinM6) [reply p2pool/util/pack.py]: 😬

File: p2pool/dash/p2p.py (hunk 2 above)
Comment (UdjinM6) [reply p2pool/dash/p2p.py]: It's just annoying to see tons of messages like "nothing to see here" when you are trying to debug smth :D But even then it should probably be in it's own PR (reverted this part).

Comment (codablock) [reply p2pool/dash/p2p.py]: :+1: btw, descriptions needs to updated as well now

**Pull request merged**